azure-keyvault-cryptography/src/main/java/com/microsoft/azure/keyvault/cryptography/algorithms/PsBase.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.keyvault.cryptography.algorithms;
+
+import java.math.BigInteger;
+import java.security.KeyPair;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.PSSParameterSpec;
+import java.util.Arrays;
+import java.util.zip.DataFormatException;
+import java.security.Signature;
+
+import com.microsoft.azure.keyvault.cryptography.ByteExtensions;
+import com.microsoft.azure.keyvault.cryptography.ISignatureTransform;
+
+/**
+ *
+ */
+public abstract class PsBase extends RsaSignature {
+
+    protected PsBase(String name) {
+        super(name);
+    }
+
+    class PsBaseSignatureTransform implements ISignatureTransform {
+
+        private final KeyPair _keyPair;
+        private final int _emLen;
+        private final int _modBits;
+        private final String _digestName;
+
+        PsBaseSignatureTransform(KeyPair keyPair, String digestName) {
+            _keyPair = keyPair;
+            BigInteger modulus = ((RSAPublicKey) _keyPair.getPublic()).getModulus();
+            _emLen = getOctetLength(modulus.bitLength());
+            _modBits = ((RSAPublicKey) _keyPair.getPublic()).getModulus().bitLength();
```

Comment:
modulus was already extracted above, so _modBits is just modulus.bitLength(). And that was used in getOctetLength as well. So an opportunity to condense this a bit.